### PR TITLE
Flush in progress2

### PR DESCRIPTION
**Pull request by debris**

### CODE REVIEW

File: src/flush/flush.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		let writer = OperationWriter::new(
+			operations.into_iter(),
+			db,
+			options.field_body_size,
+			options.external.key_index_bits
+		);
+
+		let positive_operations = writer.run()?;
+
+		let path = dir.as_ref().to_owned().join(Flush::FILE_NAME);
```
Comment (tomusdrw) [inline src/flush/flush.rs]: `to_owned()` is redundant

File: src/flush/flush.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		mmap.flush()?;
+
+		Ok(Flush {
+			path,
+			mmap,
+		})
+	}
+
+	pub fn open<P: AsRef<Path>>(dir: P) -> Result<Flush> {
+		let path = dir.as_ref().to_owned().join(Self::FILE_NAME);
```
Comment (tomusdrw) [inline src/flush/flush.rs]: `to_owned()` - redundant.

File: src/flush/flush.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		let mmap = Mmap::open_path(&path, Protection::Read)?;
+		Ok(Flush {
```
Comment (tomusdrw) [inline src/flush/flush.rs]: Checksum validation seems to be missing.

File: src/flush/iterator.rs [hunk 4]
```diff
@@ -0,0 +1,89 @@
+use byteorder::{ByteOrder, LittleEndian};
+
+#[derive(Debug, PartialEq)]
+pub struct PositiveOperation<'a> {
```
Comment (tomusdrw) [inline src/flush/iterator.rs]: Could be called "Idempotent" to denote that it can be safely applied multiple times.

Comment (debris) [reply src/flush/iterator.rs]: didn't know this name! thanks! :)

**Pull request merged**